spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/endpoint/NotificationMessageHandlerMethodArgumentResolver.java

```diff
@@ -127,7 +126,7 @@ public InputStream getBody() throws IOException {
 
 		private Charset getCharset() {
 			return this.mediaType.getCharset() != null ? this.mediaType.getCharset()
-					: Charset.forName(WebUtils.DEFAULT_CHARACTER_ENCODING);
+					: Charset.defaultCharset();
```

Comment:
This will use default charset from OS. According to SNS docs, strings are encoded with UTF-8 so perhaps instead we should use here `StandardCharsets.UTF_8`?